tiny_dnn/core/framework/tensor.h

```diff
@@ -264,83 +405,61 @@ class Tensor {
     }
 
     void *mutable_device_data() {
+        static_assert(!kConst, "Non-constant operation on constant Tensor");
         toDevice();
         data_dirty_ = true;
         return (*device_data_)();
     }
 #endif
 
     size_t size() const {
-        return host_data_.size();
+        return calcSize();
```

Comment:
ok, I see. I guess you mean when it's a View. Then we probably have a member variable `size_` that is calculated each time that `reshape()` is called.

Besides, show we also have a have variable saying whether a Tensor is a view?